test/common/config/grpc_mux_impl_test.cc

```diff
@@ -285,6 +291,39 @@ TEST_F(GrpcMuxImplTest, WatchDemux) {
   expectSendMessage(type_url, {}, "2");
 }
 
+// Verifies that warning messages get logged when Envoy detects too many requests.
+TEST_F(GrpcMuxImplTest, TooManyRequests) {
+  auto onReceiveMessage = [&](uint64_t burst, std::chrono::milliseconds delay) {
+    while (burst--) {
+      std::unique_ptr<envoy::api::v2::DiscoveryResponse> response(
```

Comment:
In general, prefer tests that aren't timing sensitive, i.e. if they were executed on an arbitrarily slow server with preemption, they wouldn't flake. I think you're probably fine here, but to the extent we can mock time out it would be useful to do so.